_inc/client/performance/search.jsx

```diff
@@ -14,59 +14,95 @@ import { withModuleSettingsFormHelpers } from 'components/module-settings/with-m
 import SettingsGroup from 'components/settings-group';
-import { getSiteAdminUrl } from 'state/initial-state';
-import { getSitePlan, isFetchingSiteData } from 'state/site';
+import { getSitePlan } from 'state/site';
 import { FormFieldset } from 'components/forms';
+import CompactFormToggle from 'components/form/form-toggle/compact';
 
-class Search extends React.Component {
-	render() {
-		const plan_is_business =
-			'is-business-plan' === getPlanClass( this.props.sitePlan.product_slug );
-		const module_enabled = this.props.getOptionValue( 'search' );
-		return (
-			<SettingsCard { ...this.props } module="search" feature={ FEATURE_SEARCH_JETPACK } hideButton>
-				<SettingsGroup
-					hasChild
-					module={ { module: 'search' } }
-					support={ {
-						text: __( 'Jetpack Search supports many customizations.' ),
-						link: 'https://jetpack.com/support/search',
-					} }
-				>
-					<p>
-						{ __(
-							'The built-in WordPress search is great for sites without much content. But as your site grows, searches slow down and return less relevant results.'
-						) }{' '}
-					</p>
-					<p>
-						{ __(
-							'Jetpack Search replaces the built-in search with a fast, scalable, customizable, and highly-relevant search hosted in the WordPress.com cloud. The result: Your users find the content they want, faster.'
-						) }{' '}
-					</p>
-					{ plan_is_business && (
+function toggleModuleFactory( { getOptionValue, isSearchPlan, toggleModuleNow, updateOptions } ) {
+	return module => {
+		toggleModuleNow( module );
+		if ( isSearchPlan && getOptionValue( 'search' ) ) {
+			updateOptions( { instant_search_enabled: true } );
+		}
+	};
+}
+
+function toggleInstantSearchFactory( { isSearchPlan, getOptionValue, updateOptions } ) {
+	return () => {
+		if ( isSearchPlan && getOptionValue( 'search' ) ) {
+			updateOptions( {
+				instant_search_enabled: ! getOptionValue( 'instant_search_enabled', 'search' ),
+			} );
+		}
+	};
+}
+
+function Search( props ) {
+	const isModuleEnabled = props.getOptionValue( 'search' );
+	const isInstantSearchEnabled = props.getOptionValue( 'instant_search_enabled', 'search' );
+
+	const toggleModule = useMemo( () => toggleModuleFactory( props ), [
+		props.getOptionValue,
+		props.isSearchPlan,
+		props.toggleModuleNow,
+		props.updateOptions,
+	] );
+	const toggleInstantSearch = useMemo( () => toggleInstantSearchFactory( props ), [
+		props.getOptionValue,
+		props.isSearchPlan,
+		props.updateOptions,
+	] );
+
+	return (
+		<SettingsCard { ...props } module="search" feature={ FEATURE_SEARCH_JETPACK } hideButton>
+			<SettingsGroup
+				hasChild
+				module={ { module: 'search' } }
+				support={ {
+					text: __( 'Jetpack Search supports many customizations.' ),
+					link: 'https://jetpack.com/support/search',
+				} }
+			>
+				<p>
+					{ __(
+						'Help visitors quickly find answers with highly relevant instant search results and powerful filtering hosted in the WordPress.com cloud.'
```

Comment:
I know this is a legacy string so feel free to ignore, but I don't know if “hosted” will convey the right message.

```suggestion
						'Help visitors quickly find answers with highly relevant instant search results and powerful filtering. Powered by the WordPress.com cloud.'
```

Reply:
Should we say "Jetpack Cloud" ?

Reply:
I thought the same thing, but Jetpack Cloud will soon mean something completely different. Though to be fair, that's not going to be the public name of the product.